### PR TITLE
Cloudflare waf integration v101

### PR DESCRIPTION
**Pull request by TalGumi**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Upgrade to version 1.0.1

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [x ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x ] No

## Must have
- [ x] Tests
- [x ] Documentation

### CODE REVIEW

**Comment by Noy-Maimon** [comment]
@TalGumi Sorry for the delay with this review, one of our developers will handle it soon, thanks!

File: Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py [hunk 1]
```diff
@@ -368,7 +376,7 @@ def cloudflare_waf_ip_list_item_delete_request(self, list_id: str, items: list)
             url_suffix=f'accounts/{self.account_id}/rules/lists/{list_id}/items',
             json_data={'items': items})
 
-    def cloudflare_waf_ip_list_item_list_request(self, list_id: str, item: list = None) -> Dict[str, Any]:
+    def cloudflare_waf_ip_list_item_list_request(self, list_id: str, item: list = None, cursors: str = None) -> Dict[str, Any]:
```
Comment (JudahSchwartz) [inline Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py:379]: please fix this docstring, the parameters arent matching up

File: Packs/CloudflareWAF/ReleaseNotes/1_0_0.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Cloudflare WAF
+- Cloudflare WAF integration allows customers to manage firewall rules, filters, and IP-lists. It also allows to retrieve zones list for each account. (Available from Cortex XSOAR 6.2.0).
```
Comment (JudahSchwartz) [inline Packs/CloudflareWAF/ReleaseNotes/1_0_0.md:4]: Please switch the notes here to be what was added in this PR only

File: Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py [hunk 3]
```diff
@@ -1034,33 +1053,54 @@ def cloudflare_waf_ip_list_item_list_command(client: Client, args: Dict[str, Any
...
 
     validate_pagination_arguments(page, page_size, limit)
 
-    item_list = client.cloudflare_waf_ip_list_item_list_request(list_id, item=item)
-    response = item_list['result']
+    ip_list = client.cloudflare_waf_ip_list_item_list_request(list_id, item=item_id)
+    response = ip_list['result']
+
+    # fetch all ip-list item, each call get 25 items
+    while dict_safe_get(ip_list, ['result_info', 'cursors', 'after']):
```
Comment (JudahSchwartz) [inline Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py:1072]: nice loop

**Review by JudahSchwartz — CHANGES_REQUESTED** [review]
Very nice PR! Some minor comments here, and we need to re-sync with master.
Thank you!

File: Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py (hunk 3 above)
Comment (TalGumi) [reply Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py:1072]: Thanks!

**Review by JudahSchwartz — COMMENTED** [review]
please note the following error in the build
 `unittests coverage should reach at least 75.88 currently 72.93`

**Review by JudahSchwartz — APPROVED** [review]

**Pull request merged**